src/varname.jl

```diff
@@ -77,27 +115,62 @@ Return the indexing tuple of the Julia variable used to generate `vn`.
 
 ```jldoctest
 julia> getindexing(@varname(x[1][2:3]))
-((1,), (2:3,))
+(@lens _[1][2:3])
 
 julia> getindexing(@varname(y))
 ()
 ```
 """
 getindexing(vn::VarName) = vn.indexing
 
+"""
+    get(obj, vn::VarName{sym})
+
+Alias for `get(obj, PropertyLens{sym}() ∘ vn.indexing)`.
+"""
+function Setfield.get(obj, vn::VarName{sym}) where {sym}
+    return Setfield.get(obj, PropertyLens{sym}() ∘ vn.indexing)
+end
+
+"""
+    set(obj, vn::VarName{sym}, value)
+
+Alias for `set(obj, PropertyLens{sym}() ∘ vn.indexing, value)`.
+"""
+function Setfield.set(obj, vn::VarName{sym}, value) where {sym}
+    return Setfield.set(obj, PropertyLens{sym}() ∘ vn.indexing, value)
+end
+
 
 Base.hash(vn::VarName, h::UInt) = hash((getsym(vn), getindexing(vn)), h)
-Base.:(==)(x::VarName, y::VarName) = getsym(x) == getsym(y) && getindexing(x) == getindexing(y)
+function Base.:(==)(x::VarName, y::VarName)
+    return getsym(x) == getsym(y) && getindexing(x) == getindexing(y)
+end
+
+# Allow compositions with lenses.
+function Base.:∘(vn::VarName{sym,<:Lens}, lens::Lens) where {sym}
+    return VarName{sym}(vn.indexing ∘ lens)
+end
 
-function Base.show(io::IO, vn::VarName)
+function Base.show(io::IO, vn::VarName{<:Any,<:Lens})
+    # No need to check `Setfield.has_atlens_support` since
+    # `VarName` does not allow dynamic lenses.
     print(io, getsym(vn))
-    for indices in getindexing(vn)
-        print(io, "[")
-        join(io, map(prettify_index, indices), ",")
-        print(io, "]")
-    end
+    _print_application(io, vn.indexing)
 end
 
+# This is all just to allow to convert `Colon()` into `:`.
+_print_application(io::IO, l::Lens) = Setfield.print_application(io, l)
+function _print_application(io::IO, l::ComposedLens)
+    _print_application(io, l.outer)
+    _print_application(io, l.inner)
+end
+_print_application(io::IO, l::IndexLens) =
+    print(io, "[", join(map(prettify_index, l.indices), ","), "]")
+# This is a bit weird but whatever. We're almost always going to
+# `concretize` anyways.
+_print_application(io::IO, l::DynamicIndexLens) = print(io, l, "(_)")
```

Comment:
Should we just remove this since we're going to warn (and possibly error) if we try to construct `VarName` using a `DynamicIndexLens`?